pkg/events/redis/redis_test.go

```diff
@@ -53,45 +55,60 @@ func redisConfig() ttnredis.Config {
 }
 
 func Example() {
+	// The task starter is used for automatic re-subscription on failure.
+	taskStarter := component.StartTaskFunc(component.DefaultStartTask)
 	// This sends all events received from Redis to the default pubsub.
-	redisPubSub := redis.WrapPubSub(events.DefaultPubSub(), ttnredis.Config{
+	redisPubSub := redis.WrapPubSub(context.TODO(), events.DefaultPubSub(), taskStarter, ttnredis.Config{
 		// Config here...
 	})
 	// Replace the default pubsub so that we will now publish to Redis.
 	events.SetDefaultPubSub(redisPubSub)
 }
 
-func TestRedisPubSub(t *testing.T) {
-	a := assertions.New(t)
+var timeout = (1 << 10) * test.Delay
 
-	events.IncludeCaller = true
+func TestRedisPubSub(t *testing.T) {
+	test.RunTest(t, test.TestConfig{
+		Timeout: 4 * timeout,
```

Comment:
note that `RunTest` sets a default (very liberal) timeout, so it mostly makes sense to set `Timeout` only to increase the value, otherwise we risk introducing flakiness on slow machines